pkg/operator/ceph/object/realm/controller.go

```diff
@@ -210,6 +211,22 @@ func (r *ReconcileObjectRealm) reconcile(request reconcile.Request) (reconcile.R
 		}
 	}
 
+	// Set the realm as default if specified and supported
+	logger.Debugf("cephObjectRealm.Spec.DefaultRealm= %t ", cephObjectRealm.Spec.DefaultRealm)
+	logger.Debugf("r.clusterInfo.CephVersion.IsAtLeast(cephver.Squid)= %t ", r.clusterInfo.CephVersion.IsAtLeast(cephver.Squid))
+	logger.Debug("r.clusterInfo.CephVersion=%t", r.clusterInfo.CephVersion.String())
+	logger.Debug("cephver.Squid=%t", cephver.Squid.String())
+	// if cephObjectRealm.Spec.DefaultRealm && r.clusterInfo.CephVersion.IsAtLeast(cephver.Squid) {
```

Comment:
Removed condition ` r.clusterInfo.CephVersion.IsAtLeast(cephver.Squid) ` because this issue #16211 
CC @travisn @BlaineEXE